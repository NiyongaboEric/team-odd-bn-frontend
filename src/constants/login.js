const loginInputs = [
  { input: 'Email Address', className: 'email', type: 'email' },
  { input: 'Password', className: 'password', type: 'password' },
];

export default loginInputs;
