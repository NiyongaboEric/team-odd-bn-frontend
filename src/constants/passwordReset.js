export const emailBox = [
  {
    input: 'Email address', className: 'email', type: 'email', title: 'email should be valid',
  },
];
export const passwordBox = [
  {
    input: 'New password',
    className: 'password',
    type: 'password',
    pattern: '(?=.*[a-z])(?=.*[A-Z])(?=.*[^a-zA-Z0-9]).{8,}',
    title:
             'A valid password should have a character, number, UPPER CASE letter and a lower case letter and should be longer than 8',
  },
  {
    input: 'Confirm password',
    className: 'confirmPassword',
    type: 'password',
    pattern: '(?=.*[a-z])(?=.*[A-Z])(?=.*[^a-zA-Z0-9]).{8,}',
    title:
             'A valid password should have a character, number, UPPER CASE letter and a lower case letter and should be longer than 8',
  },
];
