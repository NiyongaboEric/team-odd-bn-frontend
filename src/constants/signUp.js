export const signupInputs = [
  {
    input: 'Email Address', className: 'email', type: 'email',
  },
  {
    input: 'Password',
    className: 'password',
    pattern: '(?=.*[a-z])(?=.*[A-Z])(?=.*[^a-zA-Z0-9]).{8,}',
    type: 'password',
    title: 'Invalid Password',
  },
  {
    input: 'Confirm Password',
    pattern: '(?=.*[a-z])(?=.*[A-Z])(?=.*[^a-zA-Z0-9]).{8,}',
    className: 'confirmPassword',
    type: 'password',
    title: 'Invalid Password',
  },
];
export const signupFirstNameLastNameInputs = [
  { input: 'First Name', className: 'firstName', type: 'text' },
  { input: 'Last Name', className: 'lastName', type: 'text' },
];
export const extraInfoData = {
  link1: '/signin',
  link2: '/forgot-password',
  displayText1: 'Don\'t have an account',
  displayText2: 'signin here',
  displayText3: 'Forgot password?',
};
export const socialData = {
  displayGoogleText: 'Signup with google',
  displayFacebookText: 'Signup with facebook',
};
