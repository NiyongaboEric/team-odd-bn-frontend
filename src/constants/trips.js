import React from 'react';

export const tableHeads = [
  { className: 'tripIdNav', attribute: 'Trip Id', key: 'tripId' },
  { className: 'tripTypeNav', attribute: 'Trip Type', key: 'tripType' },
  { className: 'createdAtNav', attribute: 'Created At', key: 'createdAt' },
  { className: 'status', attribute: 'Status', key: 'status' },
  { className: 'actionNav', attribute: 'Action', key: 'action' },
];
export const navs = [
  { className: 'tableHeader', attribute: <h2>My Travel Requests</h2>, key: 'tableHeader' },
  { className: 'search', attribute: <input type="text" placeholder="search by date" />, key: 'search' },
];
