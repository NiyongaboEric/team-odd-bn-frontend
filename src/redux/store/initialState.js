const state = {
  auth: {
    signup: {
      signupInputData: {},
      signupData: { blockSpinner: 'blockSpinner' },
      signupError: { blockSpinner: 'blockSpinner' },
    },
  },
};
export default state;
