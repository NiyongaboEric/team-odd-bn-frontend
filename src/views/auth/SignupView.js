import React from 'react';
import dotenv from 'dotenv';
import '../../assets/css/signup.scss';
import propTypes from 'prop-types';
import SubmitButton from '../../components/SubmitButton';
import InputField from '../../components/InputField';
import initialState from '../../redux/store/index';
import {
  signupInputs,
  signupFirstNameLastNameInputs,
  socialData,
  extraInfoData,
} from '../../constants/signUp';
import CommonAuthForm from './CommonAuthForm';

dotenv.config();

const SignupView = ({ updateSignupInputAction, signupAction, stateObject }) => (
  <CommonAuthForm
    submit={signupAction}
    data={stateObject.auth.signup.signupInputData}
    extraInfoData={extraInfoData}
    socialData={socialData}
    introNote="Create account"
    components={[
      <div key="firstLastName" className="firstLastName">
        <InputField
          handleChange={updateSignupInputAction}
          inputList={signupFirstNameLastNameInputs}
        />
      </div>,
      <InputField
        key="input"
        handleChange={updateSignupInputAction}
        inputList={signupInputs}
      />,
      <SubmitButton
        key="submitButton"
        className="btn-short"
        buttonName="signup"
      />,
    ]}
  />
);
SignupView.propTypes = {
  updateSignupInputAction: propTypes.func,
  signupAction: propTypes.func,
  stateObject: propTypes.object,
};
SignupView.defaultProps = {
  updateSignupInputAction: null,
  signupAction: null,
  stateObject: initialState,
};

export default SignupView;
