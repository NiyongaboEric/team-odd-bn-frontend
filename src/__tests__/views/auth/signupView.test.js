import React from 'react';
import { shallow, mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';
import SignupView from '../../../views/auth/SignupView';
import initialState from '../../../redux/store/initialState';
import { socialData, extraInfoData } from '../../../constants/signUp';

describe('Test Sign up view page ', () => {
  it('should render', () => {
    expect(
      shallow(
        <Router>
          <SignupView
            socialData={socialData}
            extraInfoData={extraInfoData}
            stateObject={initialState}
          />
        </Router>,
      ),
    ).toMatchSnapshot();
  });

  it('should submit a form', () => {
    const formEventMocked = { preventDefault: jest.fn() };
    const signupAction = jest.fn();

    const input = mount(
      <Router>
        <SignupView
          socialData={socialData}
          extraInfoData={extraInfoData}
          signupAction={signupAction}
          stateObject={initialState}
        />
      </Router>,
    );
    const form = input.find('form');
    form.simulate('submit', formEventMocked);

    expect(signupAction.mock.calls).toBeDefined();
  });
});
