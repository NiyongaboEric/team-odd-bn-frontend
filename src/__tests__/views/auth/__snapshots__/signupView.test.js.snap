// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Sign up view page  should render 1`] = `ShallowWrapper {}`;
